pkg/controllers/workgenerator/controller.go

```diff
@@ -859,6 +896,38 @@ func buildAllWorkAppliedCondition(works map[string]*fleetv1beta1.Work, binding *
 	}
 }
 
+// buildAllWorkDiffReportedCondition builds DiffReported condition for a ClusterResourceBinding
+// by checking the DiffReported condition of all the works associated with the binding.
+func buildAllWorkDiffReportedCondition(works map[string]*fleetv1beta1.Work, binding *fleetv1beta1.ClusterResourceBinding) metav1.Condition {
```

Comment:
looks like only the binding.GetGeneration() is needed?

Reply:
Hi Ryan! The function was styled in consistence with the other two existing functions for similar purposes, `buildAllWorkAppliedCondition` and `buildAllWorkAvailableCondition`; I will update all three of them as you suggested.